### PR TITLE
Fix Build

### PR DESCRIPTION
**Pull request by bmoscon**
This updates a few things in the code, uses a newer version of ubuntu for the tests, which also uses a newer version of mongodb, and allows for use of python 3.7

Please review only - do not merge. Need to finalize what version of pandas needs to be used on 2.7

### CODE REVIEW

**Comment by bmoscon** [comment]
@jamesblackburn @shashank88 this should be good to review and merge now - take a look and let me know if you have any questions.

File: arctic/serialization/numpy_records.py [hunk 1]
```diff
@@ -253,7 +254,10 @@ def deserialize(self, item, force_bytes_to_unicode=False):
                 if len(index) and type(index[0]) == bytes:
                     index = index.astype('unicode')
 
-        return Series.from_array(data, index=index, name=name)
+        if PD_VER < '0.23.0':
```
Comment (jamesblackburn) [inline arctic/serialization/numpy_records.py:257]: Note lexicographic version comparisons don't work right in general for all versions... I guess this is OK for the pandas versions we care about?

Comment (bmoscon) [reply arctic/serialization/numpy_records.py:257]: I tested for 0.22.0, 0.23.x, 0.24.x and 1.x.x and it worked as expected

Comment (jamesblackburn) [reply arctic/serialization/numpy_records.py:257]: Ok cool. I wonder at some point if we can abstract out the pandas compat code so it doesn't need to exist everywhere, or maybe it'll all go away once we hit 1.x *hopes*

Comment (bmoscon) [reply arctic/serialization/numpy_records.py:257]: if you can get to 0.24.0 these will go away :D

**Pull request merged**